src/metabase/api/cache.clj

```diff
@@ -148,4 +150,47 @@
                              nil)))
   nil)
 
+(api/defendpoint POST "/invalidate"
+  "Invalidate cache entries"
+  [include database dashboard question]
+  {include   [:maybe [:= :overrides]]
+   database  [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]
+   dashboard [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]
+   question  [:maybe (ms/QueryVectorOf ms/IntGreaterThanOrEqualToZero)]}
+
+  (when-not (premium-features/enable-cache-granular-controls?)
+    (throw (premium-features/ee-feature-error (tru "Granular Caching"))))
+
+  (if-not (= include :overrides)
+    (let [conditions (for [[k vs] [[:database database]
+                                   [:dashboard dashboard]
+                                   [:question question]]
+                           v      vs]
+                       [:and [:= :model (name k)] [:= :model_id v]])
+          cnt        (when (seq conditions)
+                       ;; using JVM date rather than DB time since it's what are used in cache tasks
+                       (t2/query-one {:update (t2/table-name CacheConfig)
+                                      :set    {:invalidated_at (t/offset-date-time)}
+                                      :where  (into [:or] conditions)}))]
+      (if-not (and cnt (pos? cnt))
+        {:status 400
+         :body   {:message (tru "Could not find a cache configuration to invalidate.")}}
+        {:message (tru "Updated {0} cache configuration(s)." cnt)
+         :count   cnt}))
+
+    (let [card-ids (concat
+                    question
+                    (when (seq database)
+                      (t2/select-fn-vec :id [:model/Card :id] :database_id [:in database]))
+                    (when (seq dashboard)
```

Comment:
we probably want to update cache config too right?

Reply:
IDK, 1) why bother? :) 2) there is a really good chance there is no cache config pointing at exactly this starting location.